runtime/tr.source/trj9/env/PersistentCHTable.cpp

```diff
@@ -75,34 +75,16 @@ TR_PersistentCHTable::commitSideEffectGuards(TR::Compilation *comp)
 
    for (TR_ClassLoadCheck * clc = comp->getClassesThatShouldNotBeLoaded()->getFirst(); clc; clc = clc->getNext())
       {
-      for (int32_t i = 0; i < CLASSHASHTABLE_SIZE; ++i)
+      TR_OpaqueClassBlock *clazz = comp->fej9()->getClassFromSignature(clc->_name, clc->_length, comp->getCurrentMethod());
```

Comment:
Is ClassesThatShouldNotBeLoaded normalized so there is only one entry per class? If it is then this is fine. If there are duplicates we may need to cache the lookup result since the name lookup can be expensive.

Reply:
Currently, all optimizations that add to `ClassesThatShouldNotBeLoaded` will ensure duplicates are not added.

Reply:
Ok then this looks ok